src/sqlContext.js

```diff
@@ -138,6 +173,27 @@ function preparedSql( state, name, options ) {
 	function op() {
 		return prepare( statement )
 			.then( function() {
+				if ( options.stream ) {
+					cmd.stream = true;
+
+					// Can't use the lifted execute here because we need the
+					// request returned by the original callback version, which the
+					// lift would replace with a promise.
+					var req = cmd.execute( paramKeyValues, function( error ) {
+						if ( error ) {
+							throw error;
+						}
+					} );
+					var stream = new DataResultStream( req );
+					stream.on( "end", function() {
+						cmd.unprepare( function( error ) {
+							if ( error ) {
```

Comment:
Added the `stream` safeguard to `first`. Tested it with `step`, and everything works fine with streamed steps, even mixed with unstreamed steps, so I don't see any reason to prevent it.